src/App.tsx

```diff
@@ -1,9 +1,30 @@
-import React from 'react';
+import {
+  FC, useState, ChangeEventHandler, useEffect,
+} from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
+import searchTextInString from './searchTextInString';
+
+export const App: FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState(moviesFromServer);
+
+  const handleChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+    setQuery(event.target.value.trimStart());
+  };
+
+  useEffect(() => {
+    const filteredMovies = moviesFromServer.filter(movie => {
+      const isInTitle = searchTextInString(movie.title, query);
+      const isInDescription = searchTextInString(movie.description, query);
+
+      return (isInTitle || isInDescription) && movie;
```

Comment:
I'm not sure if checking for a movie is necessary, since it's an object of type Movie and this function is already using its properties (movie.title)